File: java/src/uk/co/bbc/dojo/MediaFactory.java
package uk.co.bbc.dojo;

public class MediaFactory {

    public Media create(String type, String audio, String video) {

        MediaImpl media= new MediaImpl(audio,video);

        switch (type) {
            case "simple":
                media.setAudioDecoder(new SimpleAudioDecoder());
                media.setVideoDecoder(new SimpleVideoDecoder());
                break;
            case "advancedAudio":
                media.setAudioDecoder(new AdvancedAudioDecoder());
                media.setVideoDecoder(new SimpleVideoDecoder());
                break;
            case "advancedVideo":
                media.setAudioDecoder(new SimpleAudioDecoder());
                media.setVideoDecoder(new AdvancedVideoDecoder());
                break;
            case "advancedBoth":
                media.setAudioDecoder(new AdvancedAudioDecoder());
                media.setVideoDecoder(new AdvancedVideoDecoder());
                break;
        }
        return media;
    }
}

File: scala/src/main/scala/uk/co/bbc/dojo/MediaFactory.scala
package uk.co.bbc.dojo

class MediaFactory {
  def create(mediaType: String, audio: String, video: String) = mediaType match {
    case "simple" => SimpleMedia(audio, video)
    case "advancedAudio" => AdvancedAudioMedia(audio, video)
    case "advancedVideo" => AdvancedVideoMedia(audio, video)
    case "advancedBoth" => AdvancedBothMedia(audio, video)
  }
}